tools/src/wyvern/target/corewyvernIL/expression/FieldSet.java

```diff
@@ -48,11 +54,27 @@ public void setExprToAssign(Expression exprToAssign) {
 	}
 
 	@Override
-	public ValueType typeCheck(TypeContext env) {
-		// TODO Auto-generated method stub
-		return null;
+	public ValueType typeCheck(TypeContext ctx) {
+		
+		// Figure out types of object and expression.
+		ValueType valTypeExpr = exprToAssign.typeCheck(ctx);
+		StructuralType structTypeObj = objectExpr.typeCheck(ctx).getStructuralType(ctx);
+		
+		// Figure out the type of the field.
+		DeclType declTypeField = structTypeObj.findDecl(fieldName, ctx);
+		if (!(declTypeField instanceof VarDeclType))
+			throw new RuntimeException("Type error: trying to set the field " + fieldName + ", but this isn't"
+										+ "a var declaration.");
+		ValueType valTypeField = ((VarDeclType) declTypeField).getRawResultType();
```

Comment:
Instead of getRawResultType(), use getResultType(View.from(objectExpr,ctx)).  This is important to ensure that the type is correct in the calling context--i.e. any references to the "this" variable in the type have been rewritten to use objectExpr instead.